### PR TITLE
feat: update dataset editor modal

### PR DESCRIPTION
**Pull request by lilykuang**
### SUMMARY

- reuse dataset editing modal from explore 
- convert `ChangeDatasourceModal` and `DatasourceModal`  to use hooks
- test has failed after convert to hooks but I will update test after this [pr](https://github.com/apache/incubator-superset/pull/10298) merge

Future improvement:
- convert `DatasourceEditor` to use hooks 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/views/datasetList/DatasetList.tsx [hunk 1]
```diff
@@ -517,6 +535,26 @@ const DatasetList: FunctionComponent<DatasetListProps> = ({
...
+      () => {
+        if (lastFetchDataConfig) {
+          fetchData(lastFetchDataConfig);
+        }
+        setDatasetCurrentlyEditing(null);
+        addSuccessToast(t('Updated: %s', data.table_name));
+      },
+      (err: any) => {
+        console.error(err);
+        addDangerToast(t('There was an issue Updated %s', data.table_name));
```
Comment (nytai) [inline superset-frontend/src/views/datasetList/DatasetList.tsx:553]:
```suggestion
        addDangerToast(t('There was an issue updating %s', data.table_name));
```

File: superset-frontend/src/datasource/DatasourceEditor.jsx [hunk 2]
```diff
@@ -220,8 +235,11 @@ export class DatasourceEditor extends React.PureComponent {
     this.state = {
       datasource: props.datasource,
       errors: [],
-      isDruid: props.datasource.type === 'druid',
-      isSqla: props.datasource.type === 'table',
+      isDruid:
+        props.datasource.type === 'druid' || props.datasource.type === 'druid',
```
Comment (nytai) [inline superset-frontend/src/datasource/DatasourceEditor.jsx:239]: this looks like a redundant check

**Review by nytai — APPROVED** [review]
overall LGTM, just 1 comment

**Comment by nytai** [comment]
@lilykuang Looks like cypress is failing on this test: https://github.com/preset-io/incubator-superset/blob/3445710267cef7cbbee28b2a49785723fc6d3247/superset-frontend/cypress-base/cypress/integration/explore/control.test.ts#L34

It looks like something that could have been broken with this change

File: superset-frontend/src/datasource/ChangeDatasourceModal.tsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
+/**
```
Comment (mistercrunch) [inline superset-frontend/src/datasource/ChangeDatasourceModal.tsx:1]: Not a huge deal, but let's use `git mv` to keep the history of the file attached to it! You can fix it quickly with
```bash
mv superset-frontend/src/datasource/ChangeDatasourceModal.tsx /tmp
git checkout HEAD^ superset-frontend/src/datasource/ChangeDatasourceModal.tsx
git mv superset-frontend/src/datasource/ChangeDatasourceModal.jsx superset-frontend/src/datasource/ChangeDatasourceModal.tsx
mv /tmp/ChangeDatasourceModal.tsx superset-frontend/src/datasource/
```

Comment (lilykuang) [reply superset-frontend/src/datasource/ChangeDatasourceModal.tsx:1]: I couldn't get moving history working for git

**Comment by lilykuang** [comment]
> @lilykuang Looks like cypress is failing on this test: https://github.com/preset-io/incubator-superset/blob/3445710267cef7cbbee28b2a49785723fc6d3247/superset-frontend/cypress-base/cypress/integration/explore/control.test.ts#L34
> 
> It looks like something that could have been broken with this change
@nytai The `superset/explore` page was broken after I updated put endpoint to use `api/v1/`. It have to be changed back to old endpoint for both dataset list and explore to work.

**Review by nytai — APPROVED** [review]

**Pull request merged**